### ISSUE TITLE
Crash and Warning - Last GIT Rev

### ISSUE THREAD

**Issue by samurairukasu** [rathena/rathena#474]
Last Revision has some issues:
Crash: Map Server.
![crash](https://cloud.githubusercontent.com/assets/9656445/8264993/32fc8d94-16c5-11e5-960f-228848166f7e.JPG)

And when u compile:
![warning](https://cloud.githubusercontent.com/assets/9656445/8264997/3f04b7d8-16c5-11e5-96aa-9bc101394730.JPG)

Sorry, my system is in spanish.

**Comment by aleos89** [comment]
Literally just fixed this a few moments ago in 2400262.

**Comment by samurairukasu** [comment]
Aleos, the warning and Crash still there, is not fixed. O_o

**Comment by aleos89** [comment]
Hm, I have no issues with a clean clone.
The hash above resolves the issue of the "itemdb_parse_roulette_db line 1092" that you have in your image.

For the clif_colormes() warning I just fixed it here: 44d8bd8. Thanks!

**Comment by samurairukasu** [comment]
With last 44d8bd8 in GIT the issue was fixed! Thanks, if i found something else i will tell u

Regards!!!

**Comment by samurairukasu** [comment]
Aleos, one last thing, if max packet_ver is 51 in packet_db.txt, u must increace it on src/common/utils.c
something like:

else if(date < 20140212) return 46;
	else if(date < 20140613) return 47;
	else if(date < 20141016) return 48;
	else if(date < 20141022) return 49;
	else if(date < 20141022) return 50;
	else if(date >= 20141022) return 51;

**Comment by samurairukasu** [comment]
Well, another thing, in packet_db.txt

![packet](https://cloud.githubusercontent.com/assets/9656445/8265919/066c653a-16e6-11e5-8fda-00c59285db32.JPG)

There is a " , " that separe the numbers against " : " , even for packet 50 than 51

I Must use my packets for solve that and another issues for 20141022 exe, i adapted them from hercules long time ago.

**Comment by samurairukasu** [comment]
Here i am, again jajaja, sorry XD.... I don´t what to open another issue in traker, i prefer keep reporting here, so we can make a follow of this and i can help u testing this implement for Rathena.

I don´t know if Char change sex is working that is pretended, but the command @changecharsex its seems not working, the command does what the script_command say, but only login out the account from server and when u login in again the char doesn´t change sex, male to female and viceversa.

Table Char from DB were updated with the change, adding the sex gender.

Meaby i missing something? I use 20141022 i found in hercules and patch it with nemo.

**Comment by cydh** [comment]
thank, fixed in 1278f6bcb3c464a8bec50ec5ded096f49a2b13ff
better reopen new issue. don't blame me if it'll be forgotten

**Comment by cydh** [comment]
@aleos89 
````
 src/char/char_mapif.c | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/src/char/char_mapif.c b/src/char/char_mapif.c
index 461500d..c864f07 100644
--- a/src/char/char_mapif.c
+++ b/src/char/char_mapif.c
@@ -756,7 +756,7 @@ int chmapif_parse_fwlog_changestatus(int fd){
 			char* data;
 
 			Sql_GetData(sql_handle, 0, &data, NULL); t_aid = atoi(data);
-			Sql_GetData(sql_handle, 0, &data, NULL); t_cid = atoi(data);
+			Sql_GetData(sql_handle, 1, &data, NULL); t_cid = atoi(data);
 			Sql_FreeResult(sql_handle);
 
 			if(!chlogif_isconnected())
@@ -1033,7 +1033,8 @@ int chmapif_parse_reqauth(int fd, int id){
         }
         if( runflag == CHARSERVER_ST_RUNNING && autotrade && cd ){
             uint16 mmo_charstatus_len = sizeof(struct mmo_charstatus) + 25;
-            cd->sex = sex;
+			if (cd->sex == 99)
+				cd->sex = sex;
 
             WFIFOHEAD(fd,mmo_charstatus_len);
             WFIFOW(fd,0) = 0x2afd;
@@ -1053,9 +1054,11 @@ int chmapif_parse_reqauth(int fd, int id){
             node != NULL &&
             node->account_id == account_id &&
             node->char_id == char_id &&
-            node->login_id1 == login_id1 &&
-            node->sex == sex /*&&
-            node->ip == ip*/ )
+            node->login_id1 == login_id1
+#if PACKETVER < 20141016
+            && node->sex == sex
+#endif
+            /* && node->ip == ip*/ )
         {// auth ok
             uint16 mmo_charstatus_len = sizeof(struct mmo_charstatus) + 25;
 			if (cd->sex == 99)

````
